### PR TITLE
Colony Door Standardization Update

### PR DESCRIPTION
**Pull request by PalJohnson**
# About the pull request

Reworks colony door access to be formated as opposed to being all over the place. Makes door access matter on the colony for everyone involved be it survivor or marine.

Mention if you have tested your changes. If you changed a map, make sure you used the map merge tool.
If this is an Issue Correction, you can type "Fixes Issue #169420" to link the PR to the corresponding Issue number #169420.

Remember: something that is self-evident to you might not be to others. Explain your rationale fully, even if you feel it goes without saying. -->

# Explain why it's good for the game

Simply put mappers since the departure from the SS13 way of mapping have forgotten about access and how it influences the flow of the game. The colony maps for the most part had all access to everyone and unnecessary bolting of doors to help restrict and permit flow which didn't make sense. Access requirements changed over the years and the old numbers on airlocks changed meaning even colonists didn't have access to certain areas. For example, the hallway between the bar and dormitory area in big red was inaccessible. This update if you will fixes these issues and provides a foundation for future maps as to make it easier to plan out.

A consequence of having this old system of access meant marines could travel to any area of the map without restriction. This makes it so marines don't have access to any of the colony doors which to me makes sense. Perhaps in a future update, I can make a console to grant people access to colony-specific regions much like the console on the Alaymayer in relation to USCM.

Colony door access should matter. The chaplain shouldn't have access to brig and command areas. Departments need too have a purpose and perhaps this could lead to gear only obtainable if you are in said role. Armory access to security roles or research unlocking gear in their department. Things like this can spice up maps.

**Other Changes**

- Old door access was removed due to doors inheriting access now
- Doors on the colony are no longer bolted as access requirements for doors made it absolute 

**Non-effected maps**

- New Varadero and Chances Claim did not receive this change as the map is small enough as is and would inhibit gameplay with the new survivor roles coming in
- Event-related maps
- Whiskeys Outpost

**New Survivor Access Changes**

- Colonist: (ACCESS_CIVILIAN_PUBLIC)
- Scientist: (ACCESS_CIVILIAN_PUBLIC,ACCESS_CIVILIAN_RESEARCH,ACCESS_CIVILIAN_MEDBAY)
- Doctor: (ACCESS_CIVILIAN_PUBLIC,ACCESS_CIVILIAN_RESEARCH,ACCESS_CIVILIAN_MEDBAY,ACCESS_CIVILIAN_COMMAND)
- WY Survivors: (ACCESS_CIVILIAN_PUBLIC,ACCESS_WY_CORPORATE,ACCESS_CIVILIAN_COMMAND)
- Security Survivors: (ACCESS_CIVILIAN_PUBLIC,ACCESS_CIVILIAN_BRIG,ACCESS_CIVILIAN_COMMAND)
- Chaplain: (ACCESS_CIVILIAN_PUBLIC)
- Engineer: (ACCESS_CIVILIAN_PUBLIC,ACCESS_CIVILIAN_ENGINEERING,ACCESS_CIVILIAN_LOGISTICS)
- Trucker: (ACCESS_CIVILIAN_PUBLIC,ACCESS_CIVILIAN_ENGINEERING,ACCESS_CIVILIAN_LOGISTICS)
- Interstellar Commerce Commission Liaison: (ACCESS_CIVILIAN_PUBLIC,ACCESS_WY_CORPORATE,ACCESS_CIVILIAN_COMMAND)
- Human Rights Observer: ACCESS_CIVILIAN_PUBLIC, ACCESS_CIVILIAN_COMMAND)
- Mercenary Survivors & Prisoner Survivors & Hostile Survivors; access = list(ACCESS_CIVILIAN_PUBLIC)

**New Colony Door Access Changes**

- Secure Doors (Generic): ACCESS_CIVILIAN_PUBLIC
- Security Doors: (ACCESS_CIVILIAN_BRIG, ACCESS_CIVILIAN_COMMAND)
- Maintenance Doors: (ACCESS_CIVILIAN_PUBLIC)
- Engineering Doors: (ACCESS_CIVILIAN_COMMAND, ACCESS_CIVILIAN_ENGINEERING, ACCESS_CIVILIAN_LOGISTICS)
- Medical Doors; (ACCESS_MARINE_LOGISTICS, ACCESS_MARINE_MEDBAY, ACCESS_MARINE_BRIDGE, ACCESS_CIVILIAN_PUBLIC)
- Research Doors: (ACCESS_CIVILIAN_RESEARCH, ACCESS_CIVILIAN_COMMAND, ACCESS_WY_CORPORATE)

# Testing Photographs and Procedure
[https://discord.com/channels/150315577943130112/746325498896056329/1058414131226607706](url)
[https://discord.com/channels/150315577943130112/746325498896056329/1058414147404054652](url)
[https://discord.com/channels/150315577943130112/746325498896056329/1058414168534962317](url)

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl: The_Flying_Flail
tweak: Updated access requirements on survivor roles to reflect role department
fix: Fixed previously inaccessible areas due to old access code
code: Refirnatted survivor access so that its nicer to read
mapadd: added new colony-specific doors with pre-baked access for mappers
maptweak: Removed bolted doors in favor of door access
maptweak: Updated colony access requirements for airlocks
/:cl:

### CODE REVIEW

**Comment by PalJohnson** [comment]
Not sure why the map render is not working. This is possibly due to me manually editing the lines of code in the map file. I opened all of the maps in SDMM to confirm they will open up.

**Comment by harryob** [comment]
> Not sure why the map render is not working. This is possibly due to me manually editing the lines of code in the map file. I opened all of the maps in SDMM to confirm they will open up.

its fine mdb2 works

File: code/game/machinery/doors/airlock_types.dm [hunk 1]
```diff
@@ -1,144 +1,169 @@
...
+	req_one_access = list(ACCESS_CIVILIAN_COMMAND, ACCESS_CIVILIAN_LOGISTICS)
 
-/obj/structure/machinery/door/airlock/marine
-	name = "\improper Airlock"
-	icon = 'icons/obj/structures/doors/door_marines.dmi'
-	assembly_type = /obj/structure/airlock_assembly/airlock_assembly_min
+/obj/structure/machinery/door/airlock/marine/colony
+    name = "\improper Airlock"
+    icon = 'icons/obj/structures/doors/door_marines.dmi'
+    assembly_type = /obj/structure/airlock_assembly/airlock_assembly_min
```
Comment (harryob) [inline code/game/machinery/doors/airlock_types.dm:121]: space indentation fails tests

**Review by harryob — CHANGES_REQUESTED** [review]

**Comment by Nanu308** [comment]
mapping lgtm

**Comment by PalJohnson** [comment]
I reverted changes on chances claim since that is a map with a different style of gameplay for survivors which I respect and decided it was best to leave alone.

**Pull request merged**